Sample GraphQL code (#21)
vendia/examples: features/share/graphql/creating-our-uni.md

<p align="center">
  <a href="https://vendia.net/">
    <img src="https://raw.githubusercontent.com/vendia/examples/main/vendia-logo.png" alt="vendia logo" width="100px">
  </a>
</p>

# Creating Our Universal Application

## Pre-requisites

> * [Vendia Share CLI](https://vendia.net/docs/share/cli)

Reviewer: ```suggestion
* Install the [Vendia Share CLI](https://vendia.net/docs/share/cli)
```

Author: Preference is to keep it as-is. It's consistently used throughout the examples repo. I can change it if you feel strongly but don't think it's much of a difference.

## Deploying the Uni

This example will create a Uni to store attributes regarding product inventory.

If you are not already logged in to the share service you do so by running [`share login`](https://vendia.net/docs/share/cli/commands/login):

```bash
share login
```

> The `share uni create` command can be used to deploy our Uni. You will need to copy the file `registration.json.sample` to `registration.json`. Pick a unique `name` for your uni that begins with `test-` - by default all unis share a common namespace so here is your chance to get creative. Update the `userId` attribute of each node to reflect your Vendia Share email address.

Reviewer: Why does the Uni need to begin with `test-`? Also what do you mean by "by default all unis share a common namespace so here is your chance to get creative."?

Author: We don't really define what a namespace is in our docs. This will be changed. I've also created a task to address the Uni naming recommendation.

```bash
cd uni_configuration
share uni create --config registration.json
```

> The Uni will take several minutes to deploy. We have pre-seeded several inventory items in our Uni. We can check on its status in the Vendia Share web application or with the `share` CLI.

Reviewer: ```suggestion
The Uni will take several minutes to deploy. We have seeded several inventory items in our Uni. We can check on its status in the Vendia Share web application or with the `share` CLI.
```

Author: Good call.

This example will create a Uni to store attributes regarding product inventory.

If you are not already logged in to the Share CLI, do so by running [`share login`](https://vendia.net/docs/share/cli/commands/login):

```bash
share login
```

> The `share uni create` command can be used to deploy our Uni. You will need to copy the file `registration.json.sample` to `registration.json`. Pick a unique `name` for your Uni that begins with `test-` - by default all Unis share a common namespace so here is your chance to get creative. Update the `userId` attribute of each node to reflect your Vendia Share email address.

Reviewer: ```suggestion
The `share uni create` command can be used to deploy your Uni. You will need to copy the file `registration.json.sample` to `registration.json`. Pick a unique `name` for your Uni that begins with `test-` (Unis that start with `test-` prefix can be destroyed and reused much faster than those without the `test-` prefix.)  By default, all Unis share a common namespace so here is your chance to get creative. Update the `userId` attribute of each node to reflect your Vendia Share email address.
```
